src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -69,8 +80,38 @@ public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws I
 
             @Override
             public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
-                logger.info("Created connector successfully");
-                // TODO Add the response to Global Context
+                try {
+                    logger.info("Created connector successfully");
+                    String workflowId = data.get(0).getWorkflowId();
+                    String workflowStepName = getName();
+                    ResourcesCreated newResource = new ResourcesCreated(workflowStepName, mlCreateConnectorResponse.getConnectorId());
+                    XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                    newResource.toXContent(builder, ToXContentObject.EMPTY_PARAMS);
+
+                    // The script to append a new object to the resources_created array
+                    Script script = new Script(
+                        ScriptType.INLINE,
+                        "painless",
+                        "ctx._source.resources_created.add(params.newResource)",
+                        Collections.singletonMap("newResource", newResource)
+                    );
+
+                    flowFrameworkIndicesHandler.updateFlowFrameworkSystemIndexDocWithScript(
+                        WORKFLOW_STATE_INDEX,
+                        workflowId,
+                        script,
+                        ActionListener.wrap(updateResponse -> {
+                            logger.info("updated resources craeted of {}", workflowId);
+                        },
+                            exception -> {
+                                logger.error("Failed to update workflow state with newly created resource: {}", exception.getMessage());
```

Comment:
Should be complete the future exceptionally here?

Reply:
I can try changing to that, don't have as much experience with completableFuture